### PR TITLE
feat(asm): configure ddwaf data obfuscator

### PR DESCRIPTION
**Pull request by gnufede**
## Description

Enable libddwaf data obfuscator by setting the config properly.
The configuration consists in two regexp that can be overriden by env vars.

## Checklist
- [x] Title must conform to [conventional commit](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional).
- [x] Add additional sections for `feat` and `fix` pull requests.
- [ ] Ensure tests are passing for affected code.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Motivation
ASM feature of sensitive data obfuscation

## Testing strategy
Unit tests and system tests

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] PR cannot be broken up into smaller PRs.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.
- [ ] Add to milestone.

### CODE REVIEW

File: docs/configuration.rst [hunk 1]
```diff
@@ -304,4 +304,34 @@ below:
      - True
      - Whether to enable the endpoint data collection in profiles.
 
+       .. _dd-appsec-enabled:
```
Comment (jd) [inline docs/configuration.rst:307]: those doc update seems unrelated, might be worth a different PR?

**Review by jd — APPROVED** [review]

File: docs/configuration.rst (hunk 1 above)
Comment (gnufede) [reply docs/configuration.rst:307]: Well, they're env vars we use and just found out they're not documented here, I did it in this PR because it is harmless and to avoid conflicts

**Review by avara1986 — APPROVED** [review]
